### DISCUSSION TITLE
Why have an Ungil trait that is equivalent to Send?

### DISCUSSION THREAD

**Discussion by nnmm** [PyO3/pyo3#2890] [General]
As far as I can see, `Ungil` is implemented for exactly the same types as `Send`, and also cannot be implemented for non-Send user types. Why is this trait used instead of `Send` itself? Just to express intent more clearly?

**Comment by mejrs** [comment]
That is true on stable rust, but not when using the `nightly` feature. See also https://pyo3.rs/internal/doc/pyo3/marker/
